File: 37319907-main/Week6/ProblemSet/sentimental-credit/credit.py
# TODO
from cs50 import get_int


def main():
    n = get_int("Number: ")
    # to get the first two digits of the number
    digits = finddigits(n)
   # print(f"{digit}")
    n2 = n
   # print(f"{n}")
    sum1 = 0
    sum2 = 0
    count = 0
    # do the following so that at the end we can count the the number of digits and also perform the checksum
    while True:
        digit = int(n2 % 10)
        n2 = int(n2 / 10)
        count += 1
        # if the the digit position is even
        if(count % 2 == 0):
            digit = digit * 2
            # see if number is 2 digit then convert it to the the sum of the digits
            if(digit >= 10):
                digit = (digit % 10) + 1
            sum2 = sum2 + digit
        else:
            sum1 = sum1 + digit
        # conditional statement to break out the loop as this loop is working like a do while loop
        if(n2 <= 0):
            break
    # perform the checksum
    checksum = sum1 + sum2
   # print(f"{checksum}")
   # print(f"{digits//10}")
    if(checksum % 10 == 0):
        if(count == 15):
            if(digits == 34 or digits == 37):
                print("AMEX")
            else:
                print("INVALID")
        elif(count == 16):
            if(digits == 51 or digits == 52 or digits == 53 or digits == 54 or digits == 55):
                print("MASTERCARD")
            elif(digits // 10 == 4):
                print("VISA")
            else:
                print("INVALID")
        elif(count == 13):
            if(digits // 10 == 4):
                print("VISA")
            else:
                print("INVALID")
        else:
            print("INVALID")
    else:
        print("INVALID")


def finddigits(n):
    # to find the first two digit of the number given we can convert the given number to string and use slicing techniques
    digit = int(str(n)[0:2])
    return digit


main()

File: 37319907-main/Week1/ProblemSet/credit/credit.c
#include<stdio.h>
#include<cs50.h>

int main(void)
{
    // To ask the user for givingthe input ie a credit card number
    long n = get_long("Number: ");

    // declaring a n2=n which will be used later to calculate first and second digit
    long n2 = n;
    
    // the digit variable is usedto identify/keep the individual digits of the entered number and perform operarions as required
    int digit;
    // the count variable is used to determine the position of  the digit used with respect to the last digit
    int count = 0;
    // the sum2 is the summ of (every other digit x2 starting with second to last digit)
    int sum2 = 0;
    // sum1 is the sum of the remaining digits
    int sum1 = 0;

    // loop required to save the digit and do the checksum 
    do
    {
        digit = n % 10;
        n = n / 10;
        count++;
        if (count % 2 == 0)
        {
            // to multiply every second to last digit by two and add it to a sum2
            digit = digit * 2;
            if (digit >= 10)
            {
                // to check that if any number 2x is 2 digit then add its digits individually
                digit = (digit % 10) + 1;
            }
            sum2 = sum2 + digit;
        }
        else
        {
            sum1 = sum1 + digit;
        }
    }
    while (n != 0);

    //to calculate first and second digit
    int first_digit;
    int second_digit;
    // i is a integar used to calculate the second digit in the loop
    int i = 0;
    // n3 is the inputted number
    long n3 = n2;
    do
    {
        first_digit = n2 % 10;
        n2 = n2 / 10;
        i++;
        if (i >= 2)
        {
            second_digit = n3 % 10;
            n3 = n3 / 10;
        }
    }
    while (n2 != 0);
    
    // the final result to checksum and the count and if the result we need is wrong make it INVALID
    int checksum = sum1 + sum2;
    if (checksum % 10 == 0)
    {
        if (count == 15)
        {
            if (first_digit == 3 && (second_digit == 4 || second_digit == 7))
            {
                printf("AMEX\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (count == 16)
        {
            if (first_digit == 5 && (second_digit == 1 || second_digit == 2 || second_digit == 3 || second_digit == 4 || second_digit == 5))
            {
                printf("MASTERCARD\n");
            }
            else if (first_digit == 4)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (count == 13)
        {
            if (first_digit == 4)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}